src/panels/lovelace/editor/card-editor/hui-card-preview.ts

```diff
@@ -19,6 +20,10 @@ export class HuiCardPreview extends HTMLElement {
     if (this._element) {
       this._element.hass = value;
     }
+
+    if (this._hass && computeRTL(this._hass)) {
```

Comment:
hass is updated a lot, on every state change, so you should not run this checks every time it is updated.
Run it once and then only when the language is changed.
Before setting `this._hass` check if the language is changed:
```
if (this._hass.language !== value.language) {
  this.style.direction = computeRTL(value) ? "rtl" : "ltr";
}
```